src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -8874,4 +8835,63 @@ void GatherComponentLoadsSurfAbsFact(EnergyPlusData &state)
+        return;
+    }
+    for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+        if (!state.dataSurface->SurfHasSurroundingSurfProperties(SurfNum)) continue;
+        auto &Surface = state.dataSurface->Surface(SurfNum);
+        int SrdSurfsNum = state.dataSurface->SurfSurroundingSurfacesNum(SurfNum);
+        auto &SrdSurfsProperty = state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum);
+        Real64 SrdSurfsViewFactor = 0;
+
+        SrdSurfsViewFactor += SrdSurfsProperty.SkyViewFactor;
+        SrdSurfsViewFactor += SrdSurfsProperty.GroundViewFactor;
+        for (int SrdSurfNum = 1; SrdSurfNum <= SrdSurfsProperty.TotSurroundingSurface; SrdSurfNum++) {
+            SrdSurfsViewFactor += SrdSurfsProperty.SurroundingSurfs(SrdSurfNum).ViewFactor;
+        }
+        // Check if the sum of all defined view factors > 1.0
+        if (SrdSurfsViewFactor > 1.0) {
+            ShowSevereError(state, "Illegal surrounding surfaces view factors for " + Surface.Name + ".");
+            ShowContinueError(state, " The sum of sky, ground, and all surrounding surfaces view factors should be less than or equal to 1.0.");
+        }
+        if (SrdSurfsProperty.IsSkyViewFactorSet && SrdSurfsProperty.IsGroundViewFactorSet) {
+            // If both surface sky and ground view factor defined, overwrite with the defined value
+            Surface.ViewFactorSkyIR = SrdSurfsProperty.SkyViewFactor;
+            Surface.ViewFactorGroundIR = SrdSurfsProperty.GroundViewFactor;
+        } else if (SrdSurfsProperty.IsSkyViewFactorSet && !SrdSurfsProperty.IsGroundViewFactorSet) {
+            // If only sky view factor defined, ground view factor = 1 - all other defined view factors.
+            Surface.ViewFactorSkyIR = SrdSurfsProperty.SkyViewFactor;
+            Surface.ViewFactorGroundIR = 1 - SrdSurfsViewFactor;
+            SrdSurfsProperty.GroundViewFactor = Surface.ViewFactorGroundIR;
+            SrdSurfsProperty.IsGroundViewFactorSet = true;
+        } else if (!SrdSurfsProperty.IsSkyViewFactorSet && SrdSurfsProperty.IsGroundViewFactorSet) {
+            // If only ground view factor defined, sky view factor = 1 - all other defined view factors.
+            Surface.ViewFactorGroundIR = SrdSurfsProperty.GroundViewFactor;
+            Surface.ViewFactorSkyIR = 1 - SrdSurfsViewFactor;
+            SrdSurfsProperty.SkyViewFactor = Surface.ViewFactorSkyIR;
+            SrdSurfsProperty.IsSkyViewFactorSet = true;
+        } else {
+            // If neither ground nor sky view factor specified, continue to use the original proportion.
+            Surface.ViewFactorSkyIR *= 1 - SrdSurfsViewFactor;
+            Surface.ViewFactorGroundIR *= 1 - SrdSurfsViewFactor;
+            SrdSurfsProperty.SkyViewFactor = Surface.ViewFactorSkyIR;
+            SrdSurfsProperty.GroundViewFactor = Surface.ViewFactorGroundIR;
+            SrdSurfsProperty.IsSkyViewFactorSet = true;
+            SrdSurfsProperty.IsGroundViewFactorSet = true;
```

Comment:
The `IsSkyViewFactorSet` and `IsGroundViewFactorSet` variables are reset to true once the actual view factor values are updated as shown above.